frontend/src/pages/inventory/Inventory.tsx

```diff
@@ -1,66 +1,181 @@
-import { useState, useEffect } from 'react';
+import React, { useState, useEffect } from 'react';
 import { Link } from 'react-router-dom';
-import { Button } from '@material-ui/core';
+import {
+  Button,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  IconButton,
+  Select,
+  MenuItem,
+  InputBase,
+  InputLabel
+} from '@material-ui/core';
+import { KeyboardArrowDown, KeyboardArrowUp, Search } from '@material-ui/icons/';
 
 import { Container, Card } from '../../components';
-import { API_GOOD } from '../../utils/api';
+import { API_ARCHIVE_GOOD, API_GOOD } from '../../utils/api';
 import ImportButton from './csv-impex/csv-import';
 import ExportButton from './csv-impex/csv-export';
-import './Inventory.scss';
 import { Item } from '../../interfaces/Items';
+import ItemRow from './item-row/ItemRow';
+import { useSnackbar } from '../../contexts';
+import './Inventory.scss';
+
+interface Response {
+  status: true;
+  message: string;
+}
+
+interface TableState {
+  allOpen: boolean;
+  typeFilter: number;
+  search: string;
+}
+
+function sortItemsByType(x: Item, y: Item) {
+  if (
+    (x.type === 'raw' && y.type === 'semi-finished') ||
+    (x.type === 'raw' && y.type === 'finished') ||
+    (x.type === 'semi-finished' && y.type === 'finished')
+  ) {
+    return -1;
+  } else {
+    return 1;
+  }
+}
 
 export default function Inventory() {
+  const snackbar = useSnackbar();
   const [items, setItems] = useState<Item[]>([]);
+  const [table, setTable] = useState<TableState>({
+    allOpen: false,
+    typeFilter: 0,
+    search: ''
+  });
+
+  const filters = ['None', 'Raw', 'Semi-finished', 'Finished'];
 
   const getItems = async () => {
     const request = await fetch(API_GOOD, {
       headers: { Authorization: `bearer ${localStorage.token}` }
     });
     const response = await request.json();
     const items = response.message as Item[];
-    setItems(items);
+    setItems(items.sort(sortItemsByType));
+  };
+
+  const archiveItem = async (id: number) => {
+    const data = [{ id: id, archive: true }];
+
+    const request = await fetch(API_ARCHIVE_GOOD, {
+      method: 'POST',
+      headers: {
+        Authorization: `bearer ${localStorage.token}`,
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(data)
+    });
+
+    getItems();
+
+    const responses = (await request.json()) as Response[];
+
+    // Will only run at max 1 time because we archive 1 item at a time
+    for (var i = 0; i < responses.length; i++) {
+      const response = responses[i];
+      if (!response.status) {
+        snackbar.push(response.message);
+        return;
+      }
+    }
+    snackbar.push('Archive successful.');
+  };
+
+  const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
```

Comment:
Did ``{value: number}`` not work?

Reply:
Yeah dont know why. It needs its unknown value!
![image](https://user-images.githubusercontent.com/43256461/109863806-12c35e00-7c30-11eb-8a46-890dbaff12d9.png)

Reply:
ah rip